### PR TITLE
Qorus issue 3219 changes

### PR DESCRIPTION
**Pull request by omusil24**
Not sure how to go about release notes and changing versions.

### CODE REVIEW

File: qlib/HttpServer.qm [hunk 1]
```diff
@@ -1118,6 +1149,30 @@ public class HttpServer::HttpServer inherits HttpServer::AbstractLogger {
         listeners{id}.addHandlers(h);
     }
 
+    #! remove request handler from a listener given the listener's id
+    removeHandlerFromListenerID(softstring id, string handler_name) {
```
Comment (davidnich) [inline qlib/HttpServer.qm]: change this method to return a bool if the listener was stopped or not

File: qlib/HttpServer.qm [hunk 2]
```diff
@@ -1118,6 +1149,30 @@ public class HttpServer::HttpServer inherits HttpServer::AbstractLogger {
...
+
+        if (!listeners.hasKey(id)) {
+            throw "HTTP-SERVER-ERROR", sprintf("there is no listener with ID %d", id);
+        }
+        listeners{id}.removeHandler(handler_name);
+        HttpServer::HttpListener::stopIfNoHandlers(listeners{id});
+    }
+
+    #! remove request handler from a listener given the listener's id
+    removeHandlerFromListenerID(softstring id, HttpServer::AbstractHttpRequestHandler handler) {
```
Comment (davidnich) [inline qlib/HttpServer.qm]: also here - change this method to return a bool if the listener was stopped or not

File: qlib/HttpServer.qm [hunk 3]
```diff
@@ -1118,6 +1149,30 @@ public class HttpServer::HttpServer inherits HttpServer::AbstractLogger {
         listeners{id}.addHandlers(h);
     }
 
+    #! remove request handler from a listener given the listener's id
+    removeHandlerFromListenerID(softstring id, string handler_name) {
+        lm.enter();
+        on_exit lm.exit();
+
+        if (!listeners.hasKey(id)) {
+            throw "HTTP-SERVER-ERROR", sprintf("there is no listener with ID %d", id);
```
Comment (davidnich) [inline qlib/HttpServer.qm:1184]: I think `INVALID-LISTENER` would be better

File: qlib/HttpServer.qm [hunk 4]
```diff
@@ -1118,6 +1149,30 @@ public class HttpServer::HttpServer inherits HttpServer::AbstractLogger {
...
+        HttpServer::HttpListener::stopIfNoHandlers(listeners{id});
+    }
+
+    #! remove request handler from a listener given the listener's id
+    removeHandlerFromListenerID(softstring id, HttpServer::AbstractHttpRequestHandler handler) {
+        lm.enter();
+        on_exit lm.exit();
+
+        if (!listeners.hasKey(id)) {
+            throw "HTTP-SERVER-ERROR", sprintf("there is no listener with ID %d", id);
```
Comment (davidnich) [inline qlib/HttpServer.qm:1202]: I think `INVALID-LISTENER` would be better

File: examples/test/qore/classes/HTTPClient/HTTPClient.qtest [hunk 5]
```diff
@@ -473,7 +473,20 @@ iOoM95849mZNSaYpIsF41wsZ5aQm1HhZmrKo1W8SvHhbQEOg8T+VWSUYdi8IL02k
         int port = q.get();
 
         HTTPClient hc(("url": "https://localhost:" + port));
-        assertThrows("SOCKET-CLOSED", \hc.get(), ("/"));
+        try {
```
Comment (davidnich) [inline examples/test/qore/classes/HTTPClient/HTTPClient.qtest:476]: please use the new coding indentation style when practical:
```
try {
} catch () {
    if () {
    } else {
    }
}
```

**Review by davidnich — CHANGES_REQUESTED** [review]

File: qlib/HttpServer.qm (hunk 3 above)
Comment (omusil24) [reply qlib/HttpServer.qm:1184]: Other code relies on this error being `HTTP-SERVER-ERROR`, therefore for backwards compatibility I wouldn't change this.

File: qlib/HttpServer.qm (hunk 4 above)
Comment (omusil24) [reply qlib/HttpServer.qm:1202]: Same as above

**Review by davidnich — APPROVED** [review]

**Pull request merged**